src/components/Popup/Popup.tsx

```diff
@@ -69,191 +75,203 @@ const getComputedPosition = (
  * from modals, which do take over the whole UI and are usually
  * center-screened.
  */
-export const Popup = ({
-  anchorRef,
-  animationDuration,
-  checkIfClickInside,
-  children,
-  className,
-  isVisible,
-  onAfterClose,
-  onClose,
-  position = Position.BOTTOM_CENTER,
-  showHeader,
-  title,
-  wrapperClassName,
-  zIndex
-}: PopupProps) => {
-  const handleClose = useCallback(() => {
-    onClose()
-    setTimeout(() => {
-      onAfterClose()
-    }, animationDuration)
-  }, [onClose, onAfterClose, animationDuration])
-
-  const popupRef: React.MutableRefObject<HTMLDivElement> = useClickOutside(
-    handleClose,
-    checkIfClickInside
-  )
-
-  const wrapperRef = useRef<HTMLDivElement>()
-  const originalTopPosition = useRef<number>(0)
-  const [computedPosition, setComputedPosition] = useState(position)
+export const Popup = forwardRef<HTMLElement, PopupProps>(
```

Comment:
wondering if for readability we might want to do


const PopupComponent = ({...}) => {
  return (
   )
}

export const Popup = forwardRef<HTMLElement, PopupProps>(PopupComponent)

Reply:
I changed my mind on this. When splitting it out like that, we would have to type the `ref` that is being passed to the `PopupComponents` as `MutableRefObject<HTMLElement>` for example. Also it becomes unclear where the ref is coming from

To get the name of the component to show in devtools we need to use a named function. I want to try and move away from setting properties on the function itself i.e. `.defaultProps`. Having a named function at the top level and passing it to `forwardRef` feels like it would not be consistent with the styleguide preferring arrow functions

https://github.com/AudiusProject/stems/pull/49/commits/02855b18df591e6f3fd9417c60ceef87e7b2ff9a#diff-ce687691fe458ef8d9a85cc00e06bf778a4272a39b854ce5c89e598eecb531e3R30
this is what I landed on, I think it's fairly nice because the named function is contained as an argument, we don't have to worry about coming up with another name for the "inner component", we can quickly tell where the ref is coming from, and all the types are centralized in one spot

Reply:
cool cool